src/android/jni/debug_protocol.cpp

```diff
@@ -2,152 +2,61 @@
 // Copyright(c) 2019 Intel Corporation. All Rights Reserved.
 
 #include <jni.h>
-#include <memory>
-#include <vector>
 #include "error.h"
-#include <iostream>
 #include <fstream>
-#include <string>
+#include "../../../include/librealsense2/h/rs_internal.h"
+#include "../../../common/parser.hpp"   //needed for auto completion in nGetCommands
 
-#include "../../../include/librealsense2/rs.h"
-#include "../../../include/librealsense2/hpp/rs_device.hpp"
-#include "../../api.h"
-#include "../../../common/parser.hpp"
 
-using std::string;
-using std::hex;
-
-std::vector<uint8_t> send_and_receive_raw_data(JNIEnv *env, rs2_device * dev, const std::vector<uint8_t>& input)
-{
-    std::vector<uint8_t> results;
+extern "C"
+JNIEXPORT jbyteArray JNICALL
+Java_com_intel_realsense_librealsense_DebugProtocol_nSendAndReceiveRawData(JNIEnv *env, jclass type,
+                                                                           jlong handle,
+                                                                           jbyteArray buffer_) {
+    jbyte *buffer = env->GetByteArrayElements(buffer_, NULL);
+    jsize length = env->GetArrayLength(buffer_);
 
-    rs2_error *e = NULL;
-    std::shared_ptr<const rs2_raw_data_buffer> list(
-            rs2_send_and_receive_raw_data(dev, (void*)input.data(), (uint32_t)input.size(), &e),
-            rs2_delete_raw_data);
+    rs2_error* e = NULL;
+    const rs2_raw_data_buffer* response_bytes =
+            rs2_send_and_receive_raw_data(reinterpret_cast<rs2_device*>(handle), (void*)buffer, (int)length, &e);
     handle_error(env, e);
 
-    auto size = rs2_get_raw_data_size(list.get(), &e);
+    int response_size = rs2_get_raw_data_size(response_bytes, &e);
     handle_error(env, e);
 
-    auto start = rs2_get_raw_data(list.get(), &e);
+    const unsigned char* response_start = rs2_get_raw_data(response_bytes, &e);
+    handle_error(env, e);
 
-    results.insert(results.begin(), start, start + size);
+    env->ReleaseByteArrayElements(buffer_, buffer, 0);
+    jbyteArray rv = env->NewByteArray(response_size);
+    env->SetByteArrayRegion (rv, 0, response_size, reinterpret_cast<const jbyte *>(response_start));
 
-    return results;
+    rs2_delete_raw_data(response_bytes);
+    return rv;
 }
 
-std::string hex_mode(JNIEnv *env, rs2_device * dev, const std::string& line)
+char* getFileContentFromFilePath(const char* file_path)
 {
-    std::vector<uint8_t> raw_data;
-    std::stringstream ss(line);
-    std::string word;
-    while (ss >> word)
-    {
-        std::stringstream converter;
-        int temp;
-        converter << std::hex << word;
-        converter >> temp;
-        raw_data.push_back(temp);
-    }
-    if (raw_data.empty())
-        throw std::runtime_error("Wrong input!");
+    FILE* fp;
+    long lSize;
+    char* xml_content;
 
-    auto result = send_and_receive_raw_data(env, dev, raw_data);
+    fp = fopen(file_path, "rb");
```

Comment:
Why switching from C++ to C for I/O?

Reply:
All the API should be C - the "auto" keyword and the using of this method will be corrected.

Reply:
done